SlicerProstate/SlicerProstateUtils/decorators.py

```diff
@@ -1,34 +1,40 @@
 from functools import wraps
+import itertools
 import logging
 import inspect
 import slicer
 
 
-def logmethod(level=logging.DEBUG):
+class logmethod(object):
```

Comment:
This is crazy stuff @che85! I have no idea what is going on in this class ( not after a brief look anyway!)